src/py42/services/alertrules.py

```diff
@@ -0,0 +1,136 @@
+from py42.services import BaseService
+
+
+class AlertRulesService(BaseService):
+    """A service to manage Alert Rules."""
+
+    _version = u"v1"
+    _resource = u"Rules/"
+    _api_prefix = u"/svc/api/{}/{}".format(_version, _resource)
+
+    def __init__(self, connection, user_context, user_profile_service):
+        super(AlertRulesService, self).__init__(connection)
+        self._user_context = user_context
+        self._user_profile_service = user_profile_service
+        self._exfiltration = None
+        self._cloud_share = None
+        self._file_type_mismatch = None
+
+    @property
+    def exfiltration(self):
+        if not self._exfiltration:
+            tenant_id = self._user_context.get_current_tenant_id()
+            self._exfiltration = ExfiltrationService(self._connection, tenant_id)
+        return self._exfiltration
+
+    @property
+    def cloudshare(self):
+        if not self._cloud_share:
+            tenant_id = self._user_context.get_current_tenant_id()
+            self._cloud_share = CloudShareService(self._connection, tenant_id)
+        return self._cloud_share
+
+    @property
+    def filetypemismatch(self):
+        if not self._file_type_mismatch:
+            tenant_id = self._user_context.get_current_tenant_id()
+            self._file_type_mismatch = FileTypeMismatchService(
+                self._connection, tenant_id
+            )
+        return self._file_type_mismatch
+
+    def add_user(self, rule_id, user_id):
+        tenant_id = self._user_context.get_current_tenant_id()
+        user_details = self._user_profile_service.get_by_id(user_id)
+        user_aliases = user_details["cloudUsernames"] or []
```

Comment:
Hmm... Should this be `user_details.dataget(cloudUsernames")`?

Reply:
it hasn't been a problem so far and I would say most users do not have cloud usernames -- but I do agree that it feels slightly safer.